src/RequestsTrait.php

```diff
@@ -0,0 +1,455 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Trinet\MezzioTest;
+
+use Fig\Http\Message\RequestMethodInterface;
+use Laminas\Diactoros\ServerRequest;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\UploadedFileInterface;
+use Psr\Http\Message\UriInterface;
+
+use function array_merge;
+
+trait RequestsTrait
+{
+    /**
+     * Visit the given URI with a DELETE request.
+     *
+     * @param array<string,string>                      $parsedBody
+     * @param array<string,array<string,string>|string> $headers
+     * @param array<string,string>                      $cookieParams
+     * @param array<string,string>                      $serverParams
+     */
+    public function delete(
+        UriInterface|string $uri,
+        array $parsedBody = [],
+        array $headers = [],
+        array $cookieParams = [],
+        array $serverParams = [],
+    ): ResponseInterface {
+        return $this->dispatchRequest(
+            $this->request(
```

Comment:
A typical misuse of a trait because without some [abstract members](https://www.php.net/manual/en/language.oop5.traits.php#language.oop5.traits.abstract) this will fail:

```php
(new class {
    use Trinet\MezzioTest\RequestsTrait;
})->delete('example-uri');
```

```
Fatal error: Uncaught Error: Call to undefined method class@anonymous::dispatchRequest()
```

Reply:
That's a good point, but no. 

The `traits` in this library are `@internal`, only to be used in the `MezzioTestEnvironment` class.

The user creates an instance of `MezzioTestEnvironment` and calls `get()`, `assert*()`

Here is an example, from the tests. 

https://github.com/trinet-at/mezzio-test/blob/4b63b707e4/tests/functional/MezzioTestEnvironmentTest.php#L59-L70

However, I will change it, to prevent users from attempting to "misuse" it.

Reply:
why would you need a trait then? you could just use a regular class ans instantiate it in the MezzioTestEnvironment?

Reply:
That's what I plan to do.

Temporarily using the traits to organize the methods without cluttering `MezzioTestEnvironment`.

I plan to extract each assertion into a class that implements `PHPUnit\Framework\Constraint`.